pkg/connector/role.go

```diff
@@ -183,15 +222,72 @@ func (o *roleResourceType) listSystemRoles(
 	return rv, nil
 }
 
+func (o *roleResourceType) listCustomRoles(
+	ctx context.Context,
+	_ *v2.ResourceId,
+	token *pagination.Token,
+) ([]*v2.Resource, error) {
+	_, page, err := parsePageToken(token.Token, &v2.ResourceId{ResourceType: resourceTypeRole.Id})
+	if err != nil {
+		return nil, fmt.Errorf("okta-connectorv2: failed to parse page token: %w", err)
+	}
+
+	qp := queryParams(token.Size, page)
+	roles, _, err := listOktaIamCustomRoles(ctx, o.client, token, qp)
+	if err != nil {
+		return nil, fmt.Errorf("okta-connectorv2: failed to list custom roles: %w", err)
+	}
+
+	rv := make([]*v2.Resource, 0)
+	for _, role := range roles {
+		resource, err := roleResource(ctx, role)
+		if err != nil {
+			return nil, fmt.Errorf("okta-connectorv2: failed to create role resource: %w", err)
+		}
+
+		rv = append(rv, resource)
+	}
+
+	return rv, nil
+}
+
+func listOktaIamCustomRoles(ctx context.Context, client *okta.Client, token *pagination.Token, qp *query.Params) ([]*okta.Role, *responseContext, error) {
+	url := "/api/v1/iam/roles"
```

Comment:
This should be a constant

Reply:
There is a lot of repeated code here. I would do something like 
```go
const (
	apiPathListAdministrators = "/api/internal/administrators"
	apiPathListIamCustomRoles = "/api/v1/iam/roles"
	ContentType               = "application/json"
)

func doRequest(
	ctx context.Context,
	client *okta.Client,
	path string,
	queryString string,
	target interface{},
) (*okta.Response, error) {
	reqUrl, err := url.Parse(path)
	if err != nil {
		return nil, err
	}

	if queryString != "" {
		reqUrl.RawQuery = queryString
	}

	rq := client.CloneRequestExecutor()

	req, err := rq.
		WithAccept(ContentType).
		WithContentType(ContentType).
		NewRequest(http.MethodGet, reqUrl.String(), nil)
	if err != nil {
		return nil, err
	}
	
	return rq.Do(ctx, req, &target)
}
```

Reply:
Fixed